### PR TITLE
Add installer for petsc4py and slepc4py

### PR DESCRIPTION
**Pull request by BarrySmith**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-petsc4py/package.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    """This package provides Python bindings for the PETSc package.
+    """
+    homepage = "https://pypi.python.org/pypi/petsc4py"
+    url      = "https://pypi.python.org/packages/b3/d5/84a71e3ccc13bf90b5055d264e5b256d161ae513392d0f28e8a7ac80d15c/petsc4py-3.7.0.tar.gz"
+
+    version('3.7.0', '816a20040a6a477bd637f397c9fb5b6d')
+
+    depends_on('py-setuptools', type='build')
+    depends_on('mpi')
+    depends_on('py-mpi4py')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-petsc4py/package.py]: Should this be a `type=('build', 'run')`? That's what most Python modules are.

File: var/spack/repos/builtin/packages/py-slepc4py/package.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    homepage = "https://pypi.python.org/pypi/slepc4py"
+    url      = "https://pypi.python.org/packages/b4/dd/116fbdf865f6f9eda779c5d31bc5b68f37ef3dd5dbfc3feb4aaba1565560/slepc4py-3.7.0.tar.gz"
+
+    version('3.7.0', 'c4775e88b0825c7313629c01de60ecb2')
+
+    depends_on('py-setuptools', type='build')
+    depends_on('mpi')
+    depends_on('py-mpi4py')
+    depends_on('petsc+mpi')
+    depends_on('py-petsc4py')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-slepc4py/package.py]: `py-mpi4py` and `py-petsc4py` should probably be `type=('build', 'run')`.

File: var/spack/repos/builtin/packages/py-slepc4py/package.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    url      = "https://pypi.python.org/packages/b4/dd/116fbdf865f6f9eda779c5d31bc5b68f37ef3dd5dbfc3feb4aaba1565560/slepc4py-3.7.0.tar.gz"
+
+    version('3.7.0', 'c4775e88b0825c7313629c01de60ecb2')
+
+    depends_on('py-setuptools', type='build')
+    depends_on('mpi')
+    depends_on('py-mpi4py', type=('build', 'run'))
+    depends_on('petsc+mpi')
+    depends_on('py-petsc4py', type=('build', 'run'))
+    depends_on('slepc')
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/py-slepc4py/package.py]: I'm not sure all these dependencies are needed.  For example, depending on `py-petsc4py` will already bring in `petsc`, `mpi`.

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-slepc4py/package.py]: I think it's fine to leave them in. We've toyed around with the idea of only RPATHing immediate dependencies in the past. So if `py-slepc4py` directly depends on `petsc` and `mpi`, there's no harm in making it explicit.

**Pull request merged**